File: api/src/main/java/com/mindcare/api/model/TherapyDomain.java
package com.mindcare.api.model;

import lombok.Data;
import jakarta.persistence.*;
import java.util.Set;
import java.util.HashSet;
import java.lang.Integer;
import java.time.LocalDateTime;

@Entity
@Table(name = "therapy_domain")
@Data
public class TherapyDomain {
  
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String name;

  @Column(nullable = true)
  private String description;

  @Column(nullable = false)
  private Boolean isActive = true;

  @Column(nullable = false)
  private LocalDateTime createdAt = LocalDateTime.now();

  @Column(nullable = true)
  private LocalDateTime updatedAt;

  @ManyToMany(mappedBy = "therapyDomains", fetch = FetchType.EAGER)
  private Set<UserProfessional> professionals = new HashSet<>();
}

File: api/app/Models/TherapyDomain.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class TherapyDomain extends Model
{
  protected $table = 'therapy_domains';

  protected $fillable = [
    'name',
    'description',
    'is_active'
  ];

  protected $casts = [
    'is_active' => 'boolean',
    'created_at' => 'datetime',
    'updated_at' => 'datetime'
  ];

  public function professionals(): BelongsToMany
  {
    return $this->belongsToMany(UserProfessional::class, 'professional_therapy_domain', 'therapy_domain_id', 'professional_uuid');
  }
}
